assembly/src/parsers/mod.rs

```diff
@@ -121,6 +138,13 @@ fn parse_op_token(
         _ => return Err(AssemblyError::invalid_op(op)),
     }?;
 
+    if in_debug_mode {
+        // edit the number of cycles corresponding to the asmop decorator at an index
+        if let Decorator::AsmOp(asmop_info) = &mut decorators[*dec_len].1 {
```

Comment:
`ops_len` should be available at `decorators[*dec_len].0` - right? If so, you probably don't need to track it all the way from the start of the function.

Reply:
You're right. Thanks. Will fix it.

Reply:
Fixed